### PR TITLE
User service integration into User pages

### LINKED ISSUE #284: Update User pages with appropriate api calls
References user story #82

We need our static user pages to use the user service functions to get data from the database and store it on the page.

### PR DESCRIPTION
**Pull request by AshlynAdamson**
Closes #284 

I've integrated the User service functions into their appropriate pages. Right now, the add user works, we will need to wait for the rest of the endpoints to confirm the integrations works. Wanted to get it up for review though for any possible feedback. I am confident that there won't be much more to change once the endpoints are done. I will also post a demo video once the endpoints are done.

- [x] wait for rest of user endpoints

### CODE REVIEW

File: BEIMA.Client/src/pages/Users/UserPage.js [hunk 1]
```diff
@@ -91,13 +76,26 @@ const UserPage = () => {
...
-      alert("you sure?");
+    const deleteUserCall = async () => {
+      let deleteNotif = await Notifications.warning("Warning: User Deletion", [`Are you sure you want to delete User ${username}?`]);
+      if(deleteNotif.isConfirmed){
+        let deleteResult = await DeleteUser(userId);
+        if(deleteResult.status === 200){
+          Notifications.success("User Deletion Successful", `User ${username} successfully deleted.`);
+          navigate('/users');
+        } else {
+          Notifications.error("Unable to Delete User", `Deletion of User ${username} failed.`);
```
Comment (AshlynAdamson) [inline BEIMA.Client/src/pages/Users/UserPage.js:96]: I can update this to use the message from the endpoint in the case of an admin is trying to delete the only admin account

File: BEIMA.Client/src/pages/Users/UserPage.js [hunk 2]
```diff
@@ -91,13 +76,26 @@ const UserPage = () => {
...
+        setEditable(false);
+      } else {
+        Notifications.error("Unable to Update User", `Update of User ${username} failed.`);
+      }
     }
     
-    const deleteUserCall = () => {
-      //TODO integrate with styled error/confirm message
-      alert("you sure?");
+    const deleteUserCall = async () => {
```
Comment (AshlynAdamson) [inline BEIMA.Client/src/pages/Users/UserPage.js:88]: I think having a check for if the current user is trying to delete themselves would be good. I can update this as I work on the authentication side of things

File: BEIMA.Client/src/pages/Users/UserPage.js (hunk 1 above)
Comment (josephmoore99) [reply BEIMA.Client/src/pages/Users/UserPage.js:96]: I think that would be a good idea, just to be more specific about the error. We could even make a bunch of else ifs instead if we go based off of the status code and/or message and customize our messages for each of those. But yeah, just relaying the message in the response would be the simplest way to go.

File: BEIMA.Client/src/pages/Users/AddUserPage.js [hunk 3]
```diff
@@ -37,27 +39,46 @@ const AddUserPage = () => {
...
     }
     
     if((!('Password' in passwordErrors) && !('Password Confirmation' in passwordErrors)) &&
        (userFields["Password"] !== userFields["Password Confirmation"])
       ) {
-         passwordErrors["Password"] = 'Paswords do not match';
-         passwordErrors["Password Confirmation"] = 'Paswords do not match';
+         passwordErrors["Password"] = 'Passwords do not match';
+         passwordErrors["Password Confirmation"] = 'Passwords do not match';
     }
```
Comment (josephmoore99) [inline BEIMA.Client/src/pages/Users/AddUserPage.js:55]: Let's add in checks for an 8 character password length and require special/number/uppercase characters as well

File: BEIMA.Client/src/pages/Users/UserPage.js (hunk 2 above)
Comment (josephmoore99) [reply BEIMA.Client/src/pages/Users/UserPage.js:88]: I think we should definitely prompt the user before deleting their own account, but I don't know that we should prevent them from doing so.

**Comment by AshlynAdamson** [comment]
Here's that demo

https://user-images.githubusercontent.com/58532471/161454200-01ba8183-f050-4d24-b443-a629047e4d56.mp4

**Review by josephmoore99 — APPROVED** [review]

File: BEIMA.Client/src/pages/Users/AddUserPage.js [hunk 4]
```diff
@@ -35,29 +42,53 @@ const AddUserPage = () => {
   
   function checkPassword(){
     let passwordErrors = {};
+    let requirements = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[-+_!@#$%^&*.,?]).+$");
```
Comment (roguedeleter) [inline BEIMA.Client/src/pages/Users/AddUserPage.js:45]: Should this be in a constants file?

Comment (AshlynAdamson) [reply BEIMA.Client/src/pages/Users/AddUserPage.js:45]: Yeah, that's probably a good place for it

Comment (AshlynAdamson) [reply BEIMA.Client/src/pages/Users/AddUserPage.js]: Added it to the constants file

**Review by roguedeleter — APPROVED** [review]

**Review by josephmoore99 — APPROVED** [review]

**Pull request merged**